addons/sms/models/mail_notification.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models
+from odoo.tools.translate import _
+
+
+class Notification(models.Model):
+    _inherit = 'mail.notification'
+
+    notification_type = fields.Selection(selection_add=[('sms', 'SMS')])
+    sms_id = fields.Many2one('sms.sms', string='SMS', index=True, ondelete='set null')
+    sms_number = fields.Char('SMS Number')
```

Comment:
no fields required when 'notification_type' is 'sms' ?

Reply:
Well according to spec an empty number is a "valid failure", so I suppose sms number is not really required.